### PR TITLE
Add Category Validation

### PR DESCRIPTION
**Pull request by obenacot**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [#42578](https://github.com/demisto/etc/issues/42578)

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/tests/pack_metadata_validator_test.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class TestPackMetadataValidator:
     FILES_PATH = os.path.normpath(os.path.join(__file__, f'{git_path()}/demisto_sdk/tests', 'test_files'))
 
     @pytest.mark.parametrize('metadata', [os.path.join(FILES_PATH, 'pack_metadata__valid.json'),
-                                          os.path.join(FILES_PATH, 'pack_metadata__valid__community.json'),
+                                          os.path.join(FILES_PATH, 'pack_metadata__valid__community.json')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/pack_metadata_validator_test.py:21]: revert this change

File: demisto_sdk/commands/common/tests/pack_metadata_validator_test.py [hunk 2]
```diff
@@ -30,12 +30,13 @@ def test_metadata_validator_valid(self, mocker, metadata):
...
-        os.path.join(FILES_PATH, 'pack_metadata_empty_category.json'),
-        os.path.join(FILES_PATH, 'pack_metadata_invalid_keywords.json'),
-        os.path.join(FILES_PATH, 'pack_metadata_invalid_tags.json'),
+        # os.path.join(FILES_PATH, 'pack_metadata_invalid_price.json'),
+        # os.path.join(FILES_PATH, 'pack_metadata_invalid_dependencies.json'),
+        # os.path.join(FILES_PATH, 'pack_metadata_list_dependencies.json'),
+        os.path.join(FILES_PATH, 'pack_metadata_empty_categories.json'),
+        # os.path.join(FILES_PATH, 'pack_metadata_invalid_category.json'),
+        # os.path.join(FILES_PATH, 'pack_metadata_invalid_keywords.json'),
+        # os.path.join(FILES_PATH, 'pack_metadata_invalid_tags.json'),
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/pack_metadata_validator_test.py:39]: uncomment the tests back

File: demisto_sdk/commands/common/tests/pack_metadata_validator_test.py [hunk 3]
```diff
@@ -136,3 +137,17 @@ def test_metadata_not_dict(self, mocker):
         mocker.patch.object(validator, '_add_error')
         with pytest.raises(BlockingValidationFailureException):
             assert not validator._is_pack_meta_file_structure_valid()
+
+    @pytest.mark.parametrize('metadata', [
+        os.path.join(FILES_PATH, 'pack_metadata_empty_categories.json')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/pack_metadata_validator_test.py:142]: why do you use `parametrize` for a single case?

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -1418,6 +1419,11 @@ def pack_metadata_non_approved_tags(non_approved_tags: set) -> str:
     def pack_metadata_price_change(old_price, new_price) -> str:
         return f"The pack price was changed from {old_price} to {new_price} - revert the change"
 
+    @staticmethod
+    @error_code_decorator
+    def pack_metadata_missing_categories(pack_meta_file) -> str:
+        return f'{pack_meta_file} - Missing categories'
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:1425]: improve error message

File: demisto_sdk/commands/common/errors.py [hunk 5]
```diff
@@ -238,6 +238,7 @@
     "pack_metadata_long_description": {'code': "PA126", 'ui_applicable': False, 'related_field': ''},
     "metadata_url_invalid": {'code': "PA127", 'ui_applicable': False, 'related_field': ''},
     "required_pack_file_does_not_exist": {'code': "PA128", 'ui_applicable': False, 'related_field': ''},
+    "pack_metadata_missing_categories": {'code': "PA129", 'ui_applicable': False, 'related_field': ''},
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:241]: add the error to the `ALLOWED_IGNORE` list at the top of the file

**Review by yaakovi — COMMENTED** [review]
why did you rename the test file?

File: demisto_sdk/commands/common/tests/pack_metadata_validator_test.py (hunk 3 above)
Comment (obenacot) [reply demisto_sdk/commands/common/tests/pack_metadata_validator_test.py:142]: Matter of convenience... `FILES_PATH` is a class variable so the alternative is define a local variable inside the function, using self.__class__.FILES_PATH. I replaced it anyway hope It's fine

**Review by yaakovi — APPROVED** [review]
